app/uk/gov/hmrc/exports/metrics/ExportsMetrics.scala

```diff
@@ -18,23 +18,133 @@ package uk.gov.hmrc.exports.metrics
+
+  def incrementCounter(counter: Counter): Unit = registry.counter(counter.path).inc()
+
+  def timeCall[A](timer: Timer)(block: => A): A = {
+    val timerContext: Context = startTimer(timer)
+    val result = block
+    timerContext.stop()
+    result
+  }
+
+  def timeAsyncCall[A](timer: Timer)(block: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    val timerContext: Context = startTimer(timer)
+    val timerRunning: AtomicBoolean = new AtomicBoolean(true)
+
+    try {
+      val result = block
+
+      result.foreach { _ =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+        }
+      }
+
+      result.failed.foreach { _ =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+        }
+      }
+
+      result
+    } catch {
+      case NonFatal(e) =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+        }
+        throw e
+    }
+  }
+
+  def timeAsyncCall[A](monitor: Monitor)(block: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    incrementCounter(monitor.callCounter)
+    val timerContext: Context = startTimer(monitor.timer)
+    val timerRunning: AtomicBoolean = new AtomicBoolean(true)
 
-  val counters = Map(notificationMetric -> metrics.defaultRegistry.counter(s"$notificationMetric.counter"))
+    try {
+      val result = block
 
-  def startTimer(feature: String): Context = timers(feature).time()
+      result.foreach { _ =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+          incrementCounter(monitor.completionCounter)
+        }
+      }
 
-  def incrementCounter(feature: String): Unit = counters(feature).inc()
+      result.failed.foreach { _ =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+          incrementCounter(monitor.failureCounter)
+        }
+      }
+
+      result
+    } catch {
+      case NonFatal(e) =>
+        if (timerRunning.compareAndSet(true, false)) {
+          timerContext.stop()
+          incrementCounter(monitor.failureCounter)
+        }
+        throw e
+    }
+  }
 
 }
 
-object MetricIdentifiers {
-  val notificationMetric = "submission.notification"
+object ExportsMetrics {
+
+  case class Meter(name: String) { val path = s"$name.rate" }
+  case class Timer(name: String) { val path = s"$name.timer" }
+  case class Counter(name: String) { val path = s"$name.counter" }
+
+  trait Monitor {
+    val name: String
+
+    val path: String = s"$name.monitor"
+    val timer: Timer = Timer(path)
+    val callCounter: Counter = Counter(path)
+    val completionCounter: Counter = Counter(s"$path.success")
+    val failureCounter: Counter = Counter(s"$path.failed")
+  }
+
+  object Timers {
+    val notificationTimer: Timer = Timer(notificationMetric)
+
+    val upstreamCustomsDeclarationsTimer: Timer = Timer(upstreamCustomsDeclarationsMetric)
```

Comment:
Maybe 'metricsUpstreamDeclarationSubmission' would be better to understand its purpose?